### PR TITLE
clarify findOne<and...> documentation

### PR DESCRIPTION
**Pull request by terakilobyte**
@tkaye407 Please review. This clarifies how the findOne and findOneAnd... methods behave

### CODE REVIEW

File: Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Internal/CoreRemoteMongoCollection.swift [hunk 1]
```diff
@@ -335,8 +342,12 @@ public class CoreRemoteMongoCollection<T: Codable>: Closable {
     }
 
     /**
-     * Finds a document in this collection which matches the provided filter and
-     * replaces it with the given document.
+     * Overwrites a single document in a collection based on a query filter and
+     * return the document in either its pre-replacement or post-replacement
```
Comment (tkaye407) [inline Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Internal/CoreRemoteMongoCollection.swift]: * returns

File: Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Internal/CoreRemoteMongoCollection.swift [hunk 2]
```diff
@@ -355,8 +366,12 @@ public class CoreRemoteMongoCollection<T: Codable>: Closable {
     }
 
     /**
-     * Finds a document in this collection which matches the provided filter and
-     * deletes that document.
+     * Removes a single document from a collection based on a query filter and
+     * return a document with the same form as the document immediately before
```
Comment (tkaye407) [inline Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Internal/CoreRemoteMongoCollection.swift]: * returns

File: Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Internal/CoreRemoteMongoCollection.swift [hunk 3]
```diff
@@ -315,8 +318,12 @@ public class CoreRemoteMongoCollection<T: Codable>: Closable {
     }
 
     /**
-     * Finds a document in this collection which matches the provided filter and
-     * performs the given update on that document.
+     * Updates a single document in a collection based on a query filter and
+     * return the document in either its pre-update or post-update form. Unlike
```
Comment (tkaye407) [inline Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Internal/CoreRemoteMongoCollection.swift]: * returns

**Comment by terakilobyte** [comment]
@tkaye407 good catch on `return -> returns`, incorporated

**Pull request merged**